examples/huggingface/main.py

```diff
@@ -0,0 +1,322 @@
+        action="store_true",
+        required=False,
+    )
+    parser.add_argument(
+        "-lr", "--learning_rate", help="Learning rate", type=float, required=False
+    )
+    parser.add_argument(
+        "-e",
+        "--num_train_epochs",
+        help="Number of training epochs",
+        type=int,
+        required=False,
+    )
+    parser.add_argument(
+        "-b",
+        "--per_device_train_batch_size",
+        help="Per device batch size",
+        type=int,
+        required=False,
+    )
+    parser.add_argument(
+        "-opt",
+        "--optim",
+        help="Optimizer (one of: adamw_hf, adamw_torch, adamw_apex_fused, or adafactor",
+        type=str,
+        required=False,
+    )
+    parser.add_argument(
+        "-wd",
+        "--weight_decay",
+        help="Weight decay for AdamW optimizer",
+        type=float,
+        required=False,
+    )
+    parser.add_argument(
+        "-b1",
+        "--adam_beta1",
+        help="beta1 hyperparameter for AdamW optimizer",
+        type=float,
+        required=False,
+    )
+    parser.add_argument(
+        "-b2",
+        "--adam_beta2",
+        help="beta2 hyperparameter for AdamW optimizer",
+        type=float,
+        required=False,
+    )
+    parser.add_argument(
+        "-eps",
+        "--adam_epsilon",
+        help="epsilon hyperparameter for AdamW optimizer",
+        type=float,
+        required=False,
+    )
+    parser.add_argument(
+        "-log", "--logfile", help="Log file name and path", type=str, required=False
+    )
+    args = parser.parse_args()
+    return vars(args)
+
+
+def set_training_args(training_args, args):
+    for argname, argvalue in args.items():
+        if argvalue is not None:
+            setattr(training_args, argname, argvalue)
+    return training_args
+
+
+class GPUMemoryCallback(TrainerCallback):
+    def on_epoch_end(self, args, state, control, **kwargs):
+        print(
+            "GPU mem: Tot - ",
+            torch.cuda.get_device_properties(0).total_memory,
+            "res - ",
+            torch.cuda.memory_reserved(0),
+            "used - ",
+            torch.cuda.memory_allocated(0),
+        )
+
+
+def get_free_gpu():
+    for i in range(torch.cuda.device_count()):
+        gpu_procs_str = torch.cuda.list_gpu_processes(i)
+        if "no processes are running" in gpu_procs_str:
+            return i
+    return None
+
+
+@hydra.main(config_path=".", config_name="config")
+def main(cfg: DictConfig) -> float:
+    print("args", cfg)
+
+    # Get command line arguments and apply hyperparaters to training arguments
+    args = cfg.args
+
+    # Logger setup
+    if args["logfile"] is not None:
+        logfile = args["logfile"]
+    else:
+        logfile = "translation_hf.log"
+    logging.basicConfig(filename=logfile, level=logging.INFO)
+    logger = logging.getLogger()
+
+    # Get a GPU if available
+    if torch.cuda.is_available():
+        device = "cuda:0"
```

Comment:
Is this where this function was supposed to be used?
```suggestion
        device = f"cuda:{get_free_gpu()}"
```